producer_client/lib/animals/resources/cat.rb

```diff
@@ -15,7 +15,9 @@ def list
         uri = URI("#{host}/cats")
         response = Net::HTTP.get(uri)
         parsed = JSON.parse(response)
-        parsed.fetch('cats').map(&OpenStruct.method(:new))
+        parsed.fetch('data').map { |cat_data|
```

Comment:
The client just needs to be updated to support the new format. We might be close to considering a deserializer on this side of the requests, but for now manually parsing the response seems Good Enough™